app/controllers/QuestionnaireController.scala

```diff
@@ -16,110 +16,146 @@
 
 package controllers
 
-import _root_.forms.{ QuestionnaireDiversityInfoForm, QuestionnaireEducationInfoForm, QuestionnaireOccupationInfoForm }
-import config.CSRHttp
+import _root_.forms.{QuestionnaireDiversityInfoForm, QuestionnaireEducationInfoForm, QuestionnaireOccupationInfoForm}
 import connectors.ApplicationClient
 import connectors.ExchangeObjects.Questionnaire
 import models.CachedDataWithApp
-import play.api.mvc.{ Request, Result }
-import security.Roles.{ DiversityQuestionnaireRole, EducationQuestionnaireRole, OccupationQuestionnaireRole, StartQuestionnaireRole }
+import play.api.mvc.{Request, Result}
+import security.QuestionnaireRoles._
 import uk.gov.hmrc.play.http.HeaderCarrier
+import helpers.NotificationType._
 
 import scala.concurrent.Future
 import scala.language.reflectiveCalls
 
-object QuestionnaireController extends QuestionnaireController(ApplicationClient) {
-  val http = CSRHttp
-}
+object QuestionnaireController extends QuestionnaireController(ApplicationClient)
+
+class QuestionnaireController(applicationClient: ApplicationClient) extends BaseController(applicationClient) {
 
-abstract class QuestionnaireController(applicationClient: ApplicationClient) extends BaseController(applicationClient) {
+  val QuestionnaireCompleted = "questionnaire.completed"
 
   def start = CSRSecureAppAction(StartQuestionnaireRole) { implicit request =>
     implicit user =>
-      val p = user.application.progress
       Future.successful {
-        if (!p.diversityQuestionnaire && !p.educationQuestionnaire && !p.occupationQuestionnaire) {
-          Ok(views.html.questionnaire.intro(QuestionnaireDiversityInfoForm.acceptanceForm))
-        } else {
-          Ok(views.html.questionnaire.continue())
+        (QuestionnaireCompletedRole.isAuthorized(user), QuestionnaireNotStartedRole.isAuthorized(user)) match {
```

Comment:
@satishkamatam  
I think some of these "roles" feel more like statuses of the application while in other places they represent if you can perform one action or not, combining these two concepts in one is very confusing for me.
For instance we have "roles" called:
"AssistanceDetails", "PersonalDetails" that represent if you can provide assistanceDetails, etc but what action "QuestionnaireCompletedRole" represents?